### PR TITLE
Fixes for librw to enable building it using cmake

### PR DESCRIPTION
**Pull request by madebr**
Hey,

These fixes are needed to enable building librw using cmake.
It silences warnings for users of this library that might have stronger checkers available.
This pr only contains fixes to the c code.

### CODE REVIEW

File: src/base.cpp [hunk 1]
```diff
@@ -623,7 +623,7 @@ Matrix::identityError(void)
 void
 correctPathCase(char *filename)
 {
-#ifndef _WIN32
+#if !defined(_WIN32) && !defined(RW_PS2)
```
Comment (JayFoxRox) [inline src/base.cpp]: What's the motivation behind this change?

(I'm not sure if PS2 is case sensitive)

---

Also note that this is still called below (also conditional, which looks like bad design in the original code)

Comment (JayFoxRox) [reply src/base.cpp]: How does the original premake deal with this then?

Comment (madebr) [reply src/base.cpp]: premake does not deal with it. No exceptions. https://github.com/aap/librw/blob/b1c3c1dca8c5c2244ba7ecfacd50310f27a31b8e/premake5.lua#L64-L69

**Comment by aap** [comment]
Why delete the trailing commas and add parentheses? I don't like that.

**Comment by madebr** [comment]
I removed the trailing commas because, when including your lib in openrw, I got a lot of the following type of warnings:
```
/home/maarten/programming/openrw/external/librw/librw/src/gl/../rwobjects.h:44:55: warning: comma at end of enumerator list [-Wpedantic]
   SYNCOBJ          = HIERARCHYSYNCOBJ | SUBTREESYNCOBJ,
```

**Comment by aap** [comment]
Damn. I would consider that a bug the compiler though. That's an incredibly stupid warning.

**Comment by madebr** [comment]
Yeah, I understand your dissatisfaction. Another solution would be to make these enums private.

**Comment by aap** [comment]
Can't you just tell the compiler to not complain about perfectly fine C(++) code?

**Comment by madebr** [comment]
Your lib is c++98, and prior to c++11 a trailing comma after the last enumerator is not allowed.
I removed the pedantic flag for librw and that warning is gone.

File: src/base.cpp (hunk 1 above)
Comment (madebr) [reply src/base.cpp]: https://github.com/ps2dev/ps2sdk/issues/76

Comment (madebr) [reply src/base.cpp]: I don't know what changed in my development environment (apart from disabling the optimizations for PS2),
but this error has disappeared..

**Comment by aap** [comment]
Ok, that seems fair. I'm more of a C guy and I think it's been allowed since C99 there.
